src/main.c

```diff
@@ -446,10 +446,12 @@ void load_config(void)
 	if (cf_auth_type == AUTH_HBA) {
 		struct HBA *hba = hba_load_rules(cf_auth_hba_file);
 		if (hba) {
-			if (parsed_hba)
-				hba_free(parsed_hba);
+			hba_free(parsed_hba);
 			parsed_hba = hba;
 		}
+	} else {
+		hba_free(parsed_hba);
+		parsed_hba = NULL;
 	}
```

Comment:
It looks like the else block is redundant.

We could initialized  `parsed_hba`  to NULL at line 73:

`struct HBA *parsed_hba = NULL;`

But even that is not neede.

Reply:
load_config is rerun whenever pgbouncer receives a SIGHUP or RELOAD command. So this block is executed when auth_type changes from HBA to something else. It seemed reasonable to me we free parsed_hba whenever that happens, instead of keeping the structure allocated.

Reply:
Ok. Makes sense.